pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -212,13 +223,27 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 	}
 	r.clusterInfo.Context = r.opManagerContext
 
+	cephRNSList := &cephv1.CephBlockPoolRadosNamespaceList{}
+	namespaceListOpts := client.InNamespace(cephCluster.Namespace)
+	// List cephBlockPoolRadosNamespace CR based on spec.blockPoolName and spec.name
+	matchingKey := fmt.Sprintf("%s/%s", radosNamespace.Spec.BlockPoolName, cephv1.GetRadosNamespaceName(radosNamespace))
+	err = r.client.List(r.opManagerContext, cephRNSList, &client.MatchingFields{cephRNSNameIndex: matchingKey}, namespaceListOpts)
+	if client.IgnoreNotFound(err) != nil {
+		return reconcile.Result{}, radosNamespace, errors.Wrap(err, "failed to list cephBlockPoolRadosNamespace")
+	}
+
 	// DELETE: the CR was deleted
 	if !radosNamespace.GetDeletionTimestamp().IsZero() {
 		logger.Debugf("delete cephBlockPoolRadosNamespace %q", namespacedName)
 		// On external cluster, we don't delete the rados namespace, it has to be deleted manually
 		if cephCluster.Spec.External.Enable {
 			logger.Warning("external rados namespace %q deletion is not supported, delete it manually", namespacedName)
-		} else {
+		} else if len(cephRNSList.Items) <= 1 {
```

Comment:
How about logging that we are allowing deletion when more than 1 rns is found?

Reply:
added